src/collections.h

```diff
@@ -12,10 +15,29 @@ class Collection {
 private:
     std::vector<T> Data;
 public:
+    // std::vector constructor
     Collection<T>(std::vector<T> d) {
         Data = d;
     }
 
+    // std::list constructor
+    Collection<T>(std::list<T> d) {
```

Comment:
`size()` is a constant time operation so we should instantiate `Data` to the correct size.